src/preconditions/ClientPermissions.ts

```diff
@@ -113,6 +109,17 @@ export class CorePrecondition extends AllFlowsPrecondition {
 				});
 	}
 
+	private async getAvailablePermissions(interaction: ChatInputCommandInteraction | ContextMenuCommandInteraction) {
+		if (interaction.channel) {
+			if (interaction.channel.isDMBased()) return this.dmChannelPermissions;
+
+			const channel = await this.fetchChannelFromInteraction(interaction);
```

Comment:
This fetch feels redundant now

Reply:
Better safe than sorry and it's gonna come out of the cache anyway if it's there.